runtime/caml/platform.h

```diff
@@ -49,27 +49,16 @@ Caml_inline void cpu_relax(void) {
 #endif
 }
 
-/* Loads and stores with acquire and release semantics respectively */
+/* Loads and stores with acquire, release and relaxed semantics */
 
-Caml_inline uintnat atomic_load_acq(atomic_uintnat* p)
-{
-  return atomic_load_explicit(p, memory_order_acquire);
-}
-
-Caml_inline uintnat atomic_load_relaxed(atomic_uintnat* p)
-{
-  return atomic_load_explicit(p, memory_order_relaxed);
-}
-
-Caml_inline void atomic_store_rel(atomic_uintnat* p, uintnat v)
-{
-  atomic_store_explicit(p, v, memory_order_release);
-}
-
-Caml_inline void atomic_store_relaxed(atomic_uintnat* p, uintnat v)
-{
-  atomic_store_explicit(p, v, memory_order_relaxed);
-}
+#define atomic_load_acquire(p)                    \
```

Comment:
Is there a reason to go from inlined functions to macros? The functions are more robust than macros and I believe modern compilers will do the right thing. I've seen other instances in the runtime where we've replaced macros with functions.

Reply:
I believe that this is related to Guillaume's comment, `Make [the helpers] polymorphic in the pointer type`. As functions they were declared to work on `uintnat`, while the `atomic_*_explicit` operation can work on all atomic types. The patch introduces new usage of the helpers on `int`, `uint32_t`, `opcode_t *`, `struct lf_skipcell *`, etc.

Reply:
Thanks!